sdk/python/feast/offline_server.py

```diff
@@ -215,9 +217,15 @@ def offline_write_batch(self, command: dict, key: str):
         )
 
         assert len(feature_views) == 1, "incorrect feature view"
+        feature_view = feature_views[0]
+        assert_permissions(
+            resource=feature_view,
+            actions=[AuthzedAction.WRITE_OFFLINE],
+        )
+
         table = self.flights[key]
         self.offline_store.offline_write_batch(
-            self.store.config, feature_views[0], table, command["progress"]
+            self.store.config, feature_view, table, command["progress"]
```

Comment:
since assert_permissions returns the input resource, you can embed it in the call like:
```
self.offline_store.offline_write_batch(
            self.store.config, assert_permissions(
                feature_view, ...), 
                ...
```
IMO we need to minimize and simplify the changes to introduce these permission validations

Reply:
The easy solution is to add the same `cast(FeatureView, assert...` thait we used when invoking `permitted_resources`

Reply:
done